### PR TITLE
fix shellcheck failures on vet.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
/kind cleanup

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/vet.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ done
 
 if [[ ${#targets[@]} -eq 0 ]]; then
   # Do not run on third_party directories or generated client code or build tools.
-  targets=$(go list -e ./... | egrep -v "/(build|third_party|vendor|staging|clientset_generated)/")
+  mapfile -t targets < <(go list -e ./... | grep -E -v "/(build|third_party|vendor|staging|clientset_generated)/")
```
Comment (xichengliudui) [inline hack/make-rules/vet.sh]: Ref: https://github.com/koalaman/shellcheck/wiki/SC2207

Comment (cblecker) [reply hack/make-rules/vet.sh]: We prefer not to use mapfile, as it's only bash 4+.

Comment (xichengliudui) [reply hack/make-rules/vet.sh]: It has been updated, please approve if it is correct.

File: hack/make-rules/vet.sh [hunk 2]
```diff
@@ -47,7 +47,9 @@ done
 
 if [[ ${#targets[@]} -eq 0 ]]; then
   # Do not run on third_party directories or generated client code or build tools.
-  targets=$(go list -e ./... | egrep -v "/(build|third_party|vendor|staging|clientset_generated)/")
+  while IFS=$'\n' read -r line; do
```
Comment (cblecker) [inline hack/make-rules/vet.sh]:
```suggestion
  while IFS='' read -r line; do
```

File: hack/make-rules/vet.sh [hunk 3]
```diff
@@ -47,7 +47,9 @@ done
...
+    targets=( "$line" )
```
Comment (cblecker) [inline hack/make-rules/vet.sh]:
```suggestion
    targets+=("${line}")
```

**Review by cblecker — CHANGES_REQUESTED** [review]

**Review by cblecker — APPROVED** [review]
/lgtm
/approve

**Pull request merged**